cmd/minikube/cmd/start.go

```diff
@@ -191,11 +190,12 @@ func runStart(cmd *cobra.Command, args []string) {
 		}
 	}
 
-	starter, err := provisionWithDriver(cmd, ds, existing)
+	starter, clusterDeleted, err := provisionWithDriver(cmd, ds, existing, deleteExistingCluster)
```

Comment:
We currently have something called a [FailFastError](https://github.com/kubernetes/minikube/blob/231bf12cc40c19d815e7b61f066cb259cfd7f6a0/pkg/minikube/bootstrapper/kubeadm/errors.go#L32). When this error is returned, we know that we should just exit out of minikube instead of trying to restart the cluster.

Instead of passing in `deleteExistingCluster` as a new field to the function, could we follow the above pattern and create a `DeleteClusterError`? We then check in the appropriate spot if the error we have is of the type, and if it is, then we delete the cluster.

example of fail fast error for reference:

https://github.com/kubernetes/minikube/blob/d7c517471dcfe47d4f481f8cbc3c3f0abbbe3116/pkg/minikube/node/start.go#L386

Reply:
Hey @priyawadhwa thanks for pitching in to help.

Agree. we must use the existing mechanism. 
Also, all I tried achieve here is user should have valid running cluster and would be happy if that existing cluster gets deleted and the creation of new one also fails.

I am deleting it in wrong spot and that's why I have added TODO section to recreate back the old cluster with old driver if the new cluster creation with user specified driver fails - this is not correct approach recreating old cluster back again.

I am reworking on the logic around that area.

Thank you.

Reply:
I agree with @priyawadhwa I think that there is no reason for select driver to return something which is already available globally viper.Get...

Reply:
Hey @medyagh The select driver evaluates the combination of all cases of already existing profile with same driver or different driver and the user provided --driver option and/or --delete-on-failure options and sets this flag to indicate further that the --delete-on-failure has_to_be_considered or not. Thats why i refactored the code a bit more to consolidate the logic in one place.

What @priyawadhwa mentioned above is, to make use of FailFast error type which can be used to address many cases with a single action and delete the cluster appropriately.

I believe we are all syncing up now. :)

I have reworked around this logic as mentioned earlier and will send update.

Thank you.

Reply:
however, @medyagh if you have any thought on the code refactor and select driver function kindly point out the same.